diff --git a/CMakeLists.txt b/CMakeLists.txt
index 64964b1..1e631ed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,10 +31,6 @@ set (CMAKE_C_FLAGS_DEFAULT "${CMAKE_C_FLAGS}")
 set (BXF_MANGLING "none")
 
 if (MSVC)
-    if (NOT CMAKE_GENERATOR_PLATFORM)
-        set (_ARCH "x86")
-    endif ()
-
     enable_language (ASM_MASM)
     if (CMAKE_GENERATOR MATCHES "^.*Win64$" OR _ARCH MATCHES "${AMD64}")
         set (BXF_ARCH "x86_64")
